File: COCS19/ANWCC04.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	
	#ifndef ONLINE_JUDGE
    	freopen("input", "r", stdin);
    	freopen("output", "w", stdout);
	#endif

    int t,a,b,c,d,ans;
    char ach,bch,cch;
    cin>>t;

    for(int test=0;test<t;test++){
    	cin>>a>>ach>>b>>bch>>c>>cch>>d;
    	ans = a;
    	if(ach=='+') ans+=b;
    	else ans -= b;

    	if(bch=='+') ans +=c;
    	else ans -= c;

    	if(ans==d){
    		cout<<"Valid Equation"<<endl;
    	}
    	else{
    		cout<<"Invalid Equation"<<endl;
    	}
    }
}

File: COCS19/ANWCC04.rb
if File.exists?("input")
  $stdin = File.open("input")
  $stdout = File.open("output","w")
end

t = gets.to_i
t.times do
	str = gets.chomp.split("=")#split(%r{[=\+-]})
	part = eval(str[0])
	if part==str[1].to_i
		puts "Valid"
	else
		puts "Invalid"
	end
end
